RNMT-3078 Upgrade PushWoosh MABS4

EiyuuZack opened the pull request:
## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- Finishing by placing the link to the issue here preceded by either 'Closes' OR 'Fixes' -->
In order to overcome a security vulnerability detected on Android, PushWoosh is being upgraded to a version higher than 7.16.0.

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [x] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [ ] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Android
- [x] iOS

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->
Manual tests were performed following the PushWoosh test plan which will not be linked here.

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [ ] Code follows code style of this project
- [ ] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.14.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.14.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.14.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.14.1

Commit:
[PUSH-20076] fix build problem occuring when use config-file tag inside config.xml

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.1

Commit:
[PUSH-20527] fix build issue

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.15.2

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.15.2

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.16.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.16.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.17.0

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.17.0

pw-sdk-team committed:
Pushwoosh Cordova Plugin 7.17.1

https://github.com/Pushwoosh/pushwoosh-phonegap-plugin/releases/tag/7.17.1

EiyuuZack committed:
Merge branch 'master' into feat/RNMT-3078/upgrade

usernuno commented on package.json:
Should keep the -MABS4-OS suffix

usernuno commented on plugin.xml:
Should keep the -MABS4-OS suffix

usernuno commented on plugin.xml:
Is this safe? This can cause AndroidX related issues (I assume it does not since the app compiled in order to do tests).

But it can cause conflicts with the Firebase Plugin from the Experts, we already had a support case with this issue. Please make sure both plugins can coexist.

usernuno reviewed (changes requested):
Please check the feedback I provided.

EiyuuZack commented on package.json:
I was planning to do it after merging, but if the point of this pull request is to upgrade and include our custom code it makes more sense to do it now 👍

EiyuuZack committed:
chore: add suffix to version

The whole point of this issue is to upgrade the plugin from upstream
therefore it makes sense to update the version before reaching the
MABS4 branch.

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

pmaroco commented on package.json:
Yes, if the code is already customized in the development branch, the tag should contain `-MABS4-OS`

EiyuuZack committed:
build: downgrade firebase dependencies

Match those specified by https://github.com/TruewindIT/cordova-plugin-firebase/blob/1.6/plugin.xml

References https://outsystemsrd.atlassian.net/browse/RNMT-3078

EiyuuZack commented on plugin.xml:
Indeed there were conflicts with Firebase Plugin. Question here is whether we should be the ones to accommodate an unsupported plugin or they should comply with the service.

In the interest of time, because it is a given this will not be addressed in the short term, the dependencies were downgraded to match Firebase's.

usernuno commented on plugin.xml:
I agree, we just did change that to close a high priority support case.
